### PR TITLE
Fixes #9540: Create the expected reports file for ncf in system techniques instead of first log

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/9540

### CODE REVIEW

File: techniques/system/common/1.0/promises.st [hunk 1]
```diff
@@ -376,8 +376,13 @@ bundle agent initialize_ncf()
...
 
   methods:
-      "initialize ncf"        usebundle => initialization;
-      "report about ncf init" usebundle => rudder_common_reports_generic("Common", "ncf_init", "&TRACKINGKEY&", "ncf Initialization", "None", "The ncf initialization");
+      "initialize ncf"              usebundle => initialization;
+      # Force generating expected reports file for Rudder logger
+      # because the fist call to the logger may be in audit mode.
+      "create ncf expected reports" usebundle => _create_current_expected_reports_file;
+      # Aggregate the results of the two actions
+      "combine result classes"      usebundle => _classes_combine_two("ncf_init", "logger_rudder_final_resfile", "rudder_ncf_initialization")
```
Comment (amousset) [inline techniques/system/common/1.0/promises.st]: Not sure if we should create another component or aggregate the data.

These components are not really understandable, but it is questionable to lose reporting granularity on agent side.

File: techniques/system/common/1.0/promises.st [hunk 2]
```diff
@@ -376,8 +376,13 @@ bundle agent initialize_ncf()
 {
 
   methods:
-      "initialize ncf"        usebundle => initialization;
-      "report about ncf init" usebundle => rudder_common_reports_generic("Common", "ncf_init", "&TRACKINGKEY&", "ncf Initialization", "None", "The ncf initialization");
+      "initialize ncf"              usebundle => initialization;
+      # Force generating expected reports file for Rudder logger
+      # because the fist call to the logger may be in audit mode.
```
Comment (ncharles) [inline techniques/system/common/1.0/promises.st]: typo: "the first"

File: techniques/system/common/1.0/promises.st (hunk 1 above)
Comment (ncharles) [inline techniques/system/common/1.0/promises.st]: ha, this is a good question - i think both are ncf initialization, so we can surely combine them

File: techniques/system/common/1.0/promises.st [hunk 3]
```diff
@@ -376,8 +376,13 @@ bundle agent initialize_ncf()
...
+      "create ncf expected reports" usebundle => _create_current_expected_reports_file;
+      # Aggregate the results of the two actions
+      "combine result classes"      usebundle => _classes_combine_two("ncf_init", "logger_rudder_final_resfile", "rudder_ncf_initialization")
+      "report about ncf init"       usebundle => rudder_common_reports_generic("Common", "rudder_ncf_initialization", "&TRACKINGKEY&", "ncf Initialization", "None", "The ncf initialization");
 
 }
```
Comment (ncharles) [inline techniques/system/common/1.0/promises.st]: aren't you missing a change in log_rudder.cf to not call _create_current_expected_reports_file ?

Comment (ncharles) [inline techniques/system/common/1.0/promises.st]: this is correct, thank you

**Comment by amousset** [comment]
Commit modified

**Pull request merged**